pkg/probe/http/http_test.go

```diff
@@ -135,6 +154,18 @@ func TestHTTPProbeChecker(t *testing.T) {
 			health:  probe.Success,
 			accBody: "ok body",
 		},
+		{
+			handler:    headerCounterHandler,
+			reqHeaders: http.Header{},
+			health:     probe.Success,
+			accBody:    "4",
+		},
+		{
+			handler:    headerKeysNamesHandler,
+			reqHeaders: http.Header{},
+			health:     probe.Success,
+			accBody:    "Accept\nAccept-Encoding\nConnection\nUser-Agent",
```

Comment:
Interesting, that it already sends `Accept-Encoding` out of the box. I wonder if this is something we need to update in documentation.

Alternative is to remove it when not set explicitly =) Will definitely eliminate any concerns with the size of payload

Reply:
I do tend to agree with the header removal.

Just to be coherent with the previous sentence "let's not touch this to minimize changes" I'll leave it for now and just update the docs.

But anyway I'll be happy to open a new PR to remove this header (+ and the unneeded conditions) or do it in this PR :)

Reply:
let's do documentation first and then follow up on `Accept-Encoding`. It would be great if you can drive it

Reply:
The PR for the docs is including the `Accept-Encoding` header already 
https://github.com/kubernetes/website/pull/24756
 
Yes ofc. I will work on it (removing the `Accept-Encoding` header) right afterwords